src/App.tsx

```diff
@@ -1,24 +1,243 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getpreparedTodos } from './utils/PreparedTodos';
+import { TodosFilter } from './types/TodoFilter';
+import {
+  addTodos,
+  deleteTodos,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { TodoApp } from './components/TodoApp/TodoApp';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TempTodo } from './components/TempTodo/TempTodo';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
 
-const USER_ID = 0;
+const USER_ID = 11528;
 
 export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodosFilter>(TodosFilter.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [request, setRequest] = useState(true);
+  const [loadingId, setLoadingId] = useState<number[]>([]);
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+  const [todoTitle, setTitle] = useState('');
+
+  const activeTodosCount = todos
+    .filter(({ completed }) => completed === false).length;
+
+  let isSomeTodosCompleted = todos
+    .some(({ completed }) => completed === true);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(createdTodo => {
+        setTodo(createdTodo);
+        setRequest(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setRequest(false);
+      });
+
+    const timerId = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    return () => {
+      clearInterval(timerId);
+    };
+  }, []);
+
+  const handleAddTodos = (newTodo: Omit<Todo, 'id'>) => {
+    setLoadingId([0]);
+    setIsLoading(true);
+    setRequest(true);
```

Comment:
Too match states for loading, only one is enough. `loadingIds` looks better than others, as you can track what todos are loading at this moment